src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,395 @@
+def sub (v w : vector_measure α M) : vector_measure α M :=
+{ measure_of' := v - w,
+  empty' := by simp,
+  not_measurable' := λ _ hi,
+    by simp [v.not_measurable hi, w.not_measurable hi],
+  m_Union' := λ f hf₁ hf₂,
+    has_sum.sub (v.m_Union hf₁ hf₂)
+      (w.m_Union hf₁ hf₂) }
+
+instance : has_sub (vector_measure α M) := ⟨sub⟩
+
+@[simp] lemma coe_sub (v w : vector_measure α M) : ⇑(v - w) = v - w := rfl
+lemma sub_apply (v w : vector_measure α M) (i : set α) : (v - w) i = v i - w i := rfl
+
+instance : add_comm_group (vector_measure α M) :=
+function.injective.add_comm_group _ coe_injective coe_zero coe_add coe_neg coe_sub
+
+end add_comm_group
+
+variables {M : Type*} [add_comm_group M] [topological_space M]
+variables {R : Type*} [ring R] [module R M]
+variables [topological_space R] [has_continuous_smul R M]
+
+/-- Given a real number `r` and a signed measure `s`, `smul r s` is the signed
+measure corresponding to the function `r • s`. -/
+def smul (r : R) (v : vector_measure α M) : vector_measure α M :=
+{ measure_of' := r • v,
+  empty' := by simp,
+  not_measurable' := λ _ hi, by simp [v.not_measurable hi],
+  m_Union' := λ _ hf₁ hf₂,
+    has_sum.smul (v.m_Union hf₁ hf₂) }
+
+instance : has_scalar R (vector_measure α M) := ⟨smul⟩
+
+@[simp] lemma coe_smul (r : R) (v : vector_measure α M) : ⇑(r • v) = r • v := rfl
+lemma smul_apply (r : R) (v : vector_measure α M) (i : set α) :
+  (r • v) i = r • v i := rfl
+
+instance [topological_add_group M] : module R (vector_measure α M) :=
+function.injective.module R coe_fn_add_monoid_hom coe_injective coe_smul
+
+end vector_measure
+
+namespace measure
+
+/-- A finite measure coerced into a real function is a signed measure. -/
+@[simps]
+def to_signed_measure (μ : measure α) [hμ : finite_measure μ] : signed_measure α :=
+{ measure_of' := λ i : set α, if measurable_set i then (μ.measure_of i).to_real else 0,
+  empty' := by simp [μ.empty],
+  not_measurable' := λ _ hi, if_neg hi,
+  m_Union' :=
+  begin
+    intros _ hf₁ hf₂,
+    rw [μ.m_Union hf₁ hf₂, ennreal.tsum_to_real_eq, if_pos (measurable_set.Union hf₁),
+        summable.has_sum_iff],
+    { congr, ext n, rw if_pos (hf₁ n) },
+    { refine @summable_of_nonneg_of_le _ (ennreal.to_real ∘ μ ∘ f) _ _ _ _,
+      { intro, split_ifs,
+        exacts [ennreal.to_real_nonneg, le_refl _] },
+      { intro, split_ifs,
+        exacts [le_refl _, ennreal.to_real_nonneg] },
+        exact summable_measure_to_real hf₁ hf₂ },
+    { intros a ha,
+      apply ne_of_lt hμ.measure_univ_lt_top,
+      rw [eq_top_iff, ← ha, outer_measure.measure_of_eq_coe, coe_to_outer_measure],
+      exact measure_mono (set.subset_univ _) }
+  end }
+
+lemma to_signed_measure_apply_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : measurable_set i) :
+  μ.to_signed_measure i = (μ i).to_real :=
+if_pos hi
+
+@[simp] lemma zero_to_signed_measure :
+  (0 : measure α).to_signed_measure = 0 :=
```

Comment:
```suggestion
@[simp] lemma to_signed_measure_zero :
  (0 : measure α).to_signed_measure = 0 :=
```
since dot notation is still function application on the left. Same for the `add` version